### PR TITLE
KOGITO-8573: Fixed keycloack image ref in docker-compose and gatewayAPI working inside docker-compose

### PR DESCRIPTION
**Pull request by nmirasch**
jira:https://issues.redhat.com/browse/KOGITO-8573

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request checks</b>  
  Please add comment: <b>Jenkins retest this</b>

- for a <b>specific pull request check</b>  
  Please add comment: <b>Jenkins (re)run [kogito-examples] tests</b>

- for <b>quarkus branch checks</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins run quarkus-branch</b>

- for a <b>quarkus branch specific check</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins (re)run [kogito-examples] quarkus-branch</b>

- for <b>quarkus main checks</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins run quarkus-main</b>

- for a <b>specific quarkus main check</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins (re)run [kogito-examples] quarkus-main</b>

- for <b>quarkus lts checks</b>  
  Run checks against Quarkus lts branch  
  Please add comment: <b>Jenkins run quarkus-lts</b>

- for a <b>specific quarkus lts check</b>  
  Run checks against Quarkus lts branch  
  Please add comment: <b>Jenkins (re)run [kogito-examples] quarkus-lts</b>
 
- for <b>native checks</b>  
  Run native checks  
  Please add comment: <b>Jenkins run native</b>

- for a <b>specific native check</b>  
  Run native checks 
  Please add comment: <b>Jenkins (re)run [kogito-examples] native</b>

- for <b>mandrel checks</b>  
  Run native checks against Mandrel image
  Please add comment: <b>Jenkins run mandrel</b>

- for a <b>specific mandrel check</b>  
  Run native checks against Mandrel image  
  Please add comment: <b>Jenkins (re)run [kogito-examples] mandrel</b>

- for <b>mandrel lts checks</b>  
  Run native checks against Mandrel image and quarkus lts branch
  Please add comment: <b>Jenkins run mandrel-lts</b>

- for a <b>specific mandrel lts check</b>  
  Run native checks against Mandrel image and quarkus lts branch 
  Please add comment: <b>Jenkins (re)run [kogito-examples] mandrel-lts</b>
 
</details>

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

File: kogito-quarkus-examples/process-usertasks-quarkus-with-console/src/main/resources/application.properties [hunk 1]
```diff
@@ -35,6 +35,8 @@ mp.messaging.outgoing.kogito-variables-events.value.serializer=org.apache.kafka.
 %postgresql.quarkus.datasource.jdbc.url=jdbc:postgresql://localhost:5432/kogito
 %postgresql.kogito.persistence.auto.ddl=true
 %postgresql.quarkus.kogito.devservices.enabled=false
+%postgresql.quarkus.hibernate-orm.database.generation=update
```
Comment (cristianonicolai) [inline kogito-quarkus-examples/process-usertasks-quarkus-with-console/src/main/resources/application.properties]: Flyway should handle all DB now
```suggestion
```

**Review by cristianonicolai — APPROVED** [review]

**Review by wmedvede — APPROVED** [review]
LGTM

**Review by radtriste — APPROVED** [review]

**Comment by cristianonicolai** [comment]
Jenkins rerun kogito-examples tests

**Comment by cristianonicolai** [comment]
Jenkins rerun kogito-examples tests

**Comment by cristianonicolai** [comment]
Jenkins rerun kogito-examples tests

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Comment by nmirasch** [comment]
jenkins please retest this

**Comment by kie-ci3** [comment]
**(tests) - kogito-quarkus-examples job** `#135` was: **FAILURE**
Possible explanation: Pipeline failure or project build failure

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-examples.tests.kogito-quarkus-examples/135/display/redirect
See console log:
<details>
<summary><b>Console Logs</b></summary>
GitHub pull request #1561 of commit a09c180e5e2e946daaf1fcf3b4f7d8b798c875ba, no merge conflicts.<br/>Setting status of a09c180e5e2e946daaf1fcf3b4f7d8b798c875ba to PENDING with url https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-examples.tests.kogito-quarkus-examples/135/display/redirect and message: 'Build started.'<br/>Using context: (build) kogito-quarkus-examples<br/>Checking out git https://github.com/kiegroup/kogito-pipelines.git into /var/jenkins_home/jobs/KIE/jobs/kogito/jobs/main/jobs/pullrequest/jobs/kogito-examples.tests.kogito-quarkus-examples/workspace@script/ee10b829cbd0ac763d0be6f0356db3418b2d4469e9daa1842e842313521cde92 to read .ci/jenkins/Jenkinsfile.buildchain<br/>The recommended git tool is: NONE<br/>using credential kie-ci<br/> > git rev-parse --resolve-git-dir /var/jenkins_home/jobs/KIE/jobs/kogito/jobs/main/jobs/pullrequest/jobs/kogito-examples.tests.kogito-quarkus-examples/workspace@script/ee10b829cbd0ac763d0be6f0356db3418b2d4469e9daa1842e842313521cde92/.git # timeout=10<br/>Fetching changes from the remote Git repository<br/> > git config remote.origin.url https://github.com/kiegroup/kogito-pipelines.git # timeout=10<br/>Cleaning workspace<br/> > git rev-parse --verify HEAD # timeout=10<br/>Resetting working tree<br/> > git reset --hard # timeout=10<br/> > git clean -fdx # timeout=10<br/>Fetching upstream changes from https://github.com/kiegroup/kogito-pipelines.git<br/> > git --version # timeout=10<br/> > git --version # 'git version 2.31.1'<br/>using GIT_ASKPASS to set credentials PSSWD: kie-ci service account with access to jboss-integration private github space<br/> > git fetch --tags --force --progress -- https://github.com/kiegroup/kogito-pipelines.git +refs/heads/*:refs/remotes/origin/* # timeout=10<br/> > git rev-parse origin/main^{commit} # timeout=10<br/>Checking out Revision 67b52850519066e0392935a2a839eb3e3293148b (origin/main)<br/> > git config core.sparsecheckout # timeout=10<br/> > git checkout -f 67b52850519066e0392935a2a839eb3e3293148b # timeout=10<br/>Commit message: "[KOGITO-8377]: Merged native and mandrel checks (#796)"<br/> > git rev-list --no-walk 67b52850519066e0392935a2a839eb3e3293148b # timeout=10<br/>Loading library jenkins-pipeline-shared-libraries@main<br/>Attempting to resolve main from remote references...<br/> > git --version # timeout=10<br/> > git --version # 'git version 2.31.1'<br/>using GIT_SSH to set credentials SSH: kie-ci service account with access to jboss-integration private github space<br/>[INFO] Currently running in a labeled security context<br/> > /usr/bin/chcon --type=ssh_home_t /tmp/jenkins-gitclient-ssh3393550133869653796.key<br/>Verifying host key using manually-configured host key entries<br/> > git ls-remote -h -- https://github.com/kiegroup/jenkins-pipeline-shared-libraries # timeout=10<br/>Found match: refs/heads/main revision 0dc0034954f61d52f4320c9a24f7dd83e8f79b5d<br/>The recommended git tool is: NONE<br/>using credential kie-ci-user-key<br/> > git rev-parse --resolve-git-dir /var/jenkins_home/jobs/KIE/jobs/kogito/jobs/main/jobs/pullrequest/jobs/kogito-examples.tests.kogito-quarkus-examples/workspace@libs/d03f8b53ef89e88ec2b15428a250a4b1fb7910bfcdd3e1c33e4e95c3adb758bb/.git # timeout=10<br/>Fetching changes from the remote Git repository<br/> > git config remote.origin.url https://github.com/kiegroup/jenkins-pipeline-shared-libraries # timeout=10<br/>Fetching without tags<br/>Fetching upstream changes from https://github.com/kiegroup/jenkins-pipeline-shared-libraries<br/> > git --version # timeout=10<br/> > git --version # 'git version 2.31.1'<br/>using GIT_SSH to set credentials SSH: kie-ci service account with access to jboss-integration private github space<br/>[INFO] Currently running in a labeled security context<br/> > /usr/bin/chcon --type=ssh_home_t /var/jenkins_home/jobs/KIE/jobs/kogito/jobs/main/jobs/pullrequest/jobs/kogito-examples.tests.kogito-quarkus-examples/workspace@libs/d03f8b53ef89e88ec2b15428a250a4b1fb7910bfcdd3e1c33e4e95c3adb758bb@tmp/jenkins-gitclient-ssh8749020727847766343.key<br/>Verifying host key using manually-configured host key entries<br/> > git fetch --no-tags --force --progress -- https://github.com/kiegroup/jenkins-pipeline-shared-libraries +refs/heads/*:refs/remotes/origin/* # timeout=10<br/>Checking out Revision 0dc0034954f61d52f4320c9a24f7dd83e8f79b5d (main)<br/> > git config core.sparsecheckout # timeout=10<br/> > git checkout -f 0dc0034954f61d52f4320c9a24f7dd83e8f79b5d # timeout=10<br/>Commit message: "fix build-chain version to be installed from npm (#263)"<br/> > git rev-list --no-walk 0dc0034954f61d52f4320c9a24f7dd83e8f79b5d # timeout=10<br/>[Pipeline] Start of Pipeline<br/>[Pipeline] node<br/>Still waiting to schedule task<br/>Waiting for next available executor on ‘kie-rhel7&&kie-mem16g&&!built-in’<br/>Running on rhos-01-kie-psi-rhel7-xlarge-632 in /home/jenkins/workspace/KIE/kogito/main/pullrequest/kogito-examples.tests.kogito-quarkus-examples<br/>[Pipeline] {<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Checkout SCM)<br/>[Pipeline] checkout<br/>The recommended git tool is: NONE<br/>using credential kie-ci<br/>Cloning the remote Git repository<br/>Cloning repository https://github.com/kiegroup/kogito-pipelines.git<br/> > git init /home/jenkins/workspace/KIE/kogito/main/pullrequest/kogito-examples.tests.kogito-quarkus-examples # timeout=10<br/>Fetching upstream changes from https://github.com/kiegroup/kogito-pipelines.git<br/> > git --version # timeout=10<br/> > git --version # 'git version 2.39.1'<br/>using GIT_ASKPASS to set credentials PSSWD: kie-ci service account with access to jboss-integration private github space<br/> > git fetch --tags --force --progress -- https://github.com/kiegroup/kogito-pipelines.git +refs/heads/*:refs/remotes/origin/* # timeout=10<br/>Avoid second fetch<br/>Checking out Revision 67b52850519066e0392935a2a839eb3e3293148b (origin/main)<br/>Commit message: "[KOGITO-8377]: Merged native and mandrel checks (#796)"<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] withEnv<br/>[Pipeline] {<br/>[Pipeline] withEnv<br/>[Pipeline] {<br/>[Pipeline] timestamps<br/>[Pipeline] {<br/>[Pipeline] timeout<br/>[2023-02-09T14:36:49.526Z] Timeout set to expire in 3 hr 0 min<br/>[Pipeline] {<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Tool Install)<br/>[Pipeline] tool<br/>[Pipeline] envVarsForTool<br/>[Pipeline] }<br/> > git config remote.origin.url https://github.com/kiegroup/kogito-pipelines.git # timeout=10<br/> > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10<br/> > git rev-parse origin/main^{commit} # timeout=10<br/> > git config core.sparsecheckout # timeout=10<br/> > git checkout -f 67b52850519066e0392935a2a839eb3e3293148b # timeout=10<br/>[Pipeline] // stage<br/>[Pipeline] withEnv<br/>[Pipeline] {<br/>[Pipeline] stage<br/>[Pipeline] { (Initialize)<br/>[Pipeline] tool<br/>[Pipeline] envVarsForTool<br/>[Pipeline] withEnv<br/>[Pipeline] {<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>[2023-02-09T14:36:50.161Z] + printenv<br/>[Pipeline] archiveArtifacts<br/>[2023-02-09T14:36:50.187Z] Archiving artifacts<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // withEnv<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (check space before build)<br/>[Pipeline] tool<br/>[Pipeline] envVarsForTool<br/>[Pipeline] withEnv<br/>[Pipeline] {<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] dir<br/>[2023-02-09T14:36:50.649Z] Running in /home/jenkins/workspace/KIE/kogito/main/pullrequest/kogito-examples.tests.kogito-quarkus-examples<br/>[Pipeline] {<br/>[Pipeline] echo<br/>[2023-02-09T14:36:50.685Z] [INFO] space left on the machine<br/>[Pipeline] sh<br/>[2023-02-09T14:36:50.964Z] + df -h<br/>[2023-02-09T14:36:50.965Z] Filesystem      Size  Used Avail Use% Mounted on<br/>[2023-02-09T14:36:50.965Z] devtmpfs        7.8G     0  7.8G   0% /dev<br/>[2023-02-09T14:36:50.965Z] tmpfs           7.8G     0  7.8G   0% /dev/shm<br/>[2023-02-09T14:36:50.965Z] tmpfs           7.8G   17M  7.8G   1% /run<br/>[2023-02-09T14:36:50.965Z] tmpfs           7.8G     0  7.8G   0% /sys/fs/cgroup<br/>[2023-02-09T14:36:50.965Z] /dev/vda1        80G   20G   61G  25% /<br/>[2023-02-09T14:36:50.965Z] tmpfs           1.6G     0  1.6G   0% /run/user/1000<br/>[Pipeline] echo<br/>[2023-02-09T14:36:50.987Z] [INFO] space of /home/jenkins<br/>[Pipeline] sh<br/>[2023-02-09T14:36:51.269Z] + du -h -d1 /home/jenkins<br/>[2023-02-09T14:36:51.270Z] 4.0K	/home/jenkins/.ssh<br/>[2023-02-09T14:36:51.270Z] 0	/home/jenkins/go<br/>[2023-02-09T14:36:51.270Z] 8.0K	/home/jenkins/.vnc<br/>[2023-02-09T14:36:51.270Z] 8.0K	/home/jenkins/.m2<br/>[2023-02-09T14:36:51.270Z] 2.2G	/home/jenkins/git-repos<br/>[2023-02-09T14:36:51.270Z] 4.0K	/home/jenkins/.config<br/>[2023-02-09T14:36:51.270Z] 131M	/home/jenkins/virtenvs<br/>[2023-02-09T14:36:51.270Z] 4.0K	/home/jenkins/.cekit<br/>[2023-02-09T14:36:51.270Z] 33M	/home/jenkins/remoting<br/>[2023-02-09T14:36:51.270Z] 0	/home/jenkins/.cache<br/>[2023-02-09T14:36:51.270Z] 3.8M	/home/jenkins/workspace<br/>[2023-02-09T14:36:51.270Z] 0	/home/jenkins/.pki<br/>[2023-02-09T14:36:51.270Z] 119M	/home/jenkins/tools<br/>[2023-02-09T14:36:51.270Z] 0	/home/jenkins/caches<br/>[2023-02-09T14:36:51.270Z] 17M	/home/jenkins/.npm<br/>[2023-02-09T14:36:51.270Z] 2.5G	/home/jenkins<br/>[Pipeline] echo<br/>[2023-02-09T14:36:51.293Z] [INFO] space of workspace<br/>[Pipeline] sh<br/>[2023-02-09T14:36:51.572Z] + du -h -d3 /home/jenkins/workspace<br/>[2023-02-09T14:36:51.573Z] 3.8M	/home/jenkins/workspace/KIE/kogito/main<br/>[2023-02-09T14:36:51.573Z] 3.8M	/home/jenkins/workspace/KIE/kogito<br/>[2023-02-09T14:36:51.573Z] 3.8M	/home/jenkins/workspace/KIE<br/>[2023-02-09T14:36:51.573Z] 3.8M	/home/jenkins/workspace<br/>[Pipeline] }<br/>[Pipeline] // dir<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // withEnv<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Install build-chain tool)<br/>[Pipeline] tool<br/>[Pipeline] envVarsForTool<br/>[Pipeline] withEnv<br/>[Pipeline] {<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] echo<br/>[2023-02-09T14:36:51.965Z] [INFO] Getting build-chain version from composite action file<br/>[Pipeline] readFile<br/>[Pipeline] echo<br/>[2023-02-09T14:36:52.035Z] [INFO] build-chain version recovered '^v3'<br/>[Pipeline] sh<br/>[2023-02-09T14:36:52.317Z] + npm install -g '@kie/build-chain-action@^v3' -registry=https://repository.engineering.redhat.com/nexus/repository/registry.npmjs.org/<br/>[2023-02-09T14:36:53.677Z] <br/>[2023-02-09T14:36:53.677Z] changed 74 packages, and audited 75 packages in 1s<br/>[2023-02-09T14:36:53.677Z] <br/>[2023-02-09T14:36:53.677Z] 5 packages are looking for funding<br/>[2023-02-09T14:36:53.677Z]   run `npm fund` for details<br/>[2023-02-09T14:36:53.677Z] <br/>[2023-02-09T14:36:53.677Z] found 0 vulnerabilities<br/>[Pipeline] sh<br/>[2023-02-09T14:36:53.965Z] + npm list -g<br/>[2023-02-09T14:36:53.966Z] + grep build-chain<br/>[2023-02-09T14:36:54.220Z] ├── @kie/build-chain-action@3.0.11<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // withEnv<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Build quarkus)<br/>Stage "Build quarkus" skipped due to when conditional<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Build projects)<br/>[Pipeline] tool<br/>[Pipeline] envVarsForTool<br/>[Pipeline] tool<br/>[Pipeline] envVarsForTool<br/>[Pipeline] tool<br/>[Pipeline] envVarsForTool<br/>[Pipeline] withEnv<br/>[Pipeline] {<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] echo<br/>[2023-02-09T14:36:54.795Z] BUILD_MVN_OPTS_CURRENT = -Dvalidate-formatting <br/>[Pipeline] configFileProvider<br/>[2023-02-09T14:36:54.813Z] provisioning config files...<br/>[2023-02-09T14:36:54.819Z] copy managed file [kogito_pr_settings] to file:/home/jenkins/workspace/KIE/kogito/main/pullrequest/kogito-examples.tests.kogito-quarkus-examples@tmp/config15108422816718305140tmp<br/>[Pipeline] {<br/>[Pipeline] withCredentials<br/>[2023-02-09T14:36:54.861Z] Masking supported pattern matches of $GITHUB_TOKEN<br/>[Pipeline] {<br/>[Pipeline] echo<br/>[2023-02-09T14:36:54.906Z] BUILD_MVN_OPTS =  -s /home/jenkins/workspace/KIE/kogito/main/pullrequest/kogito-examples.tests.kogito-quarkus-examples@tmp/config15108422816718305140tmp -Dmaven.wagon.http.ssl.insecure=true -Dmaven.test.failure.ignore=true<br/>[Pipeline] sh<br/>[2023-02-09T14:36:54.928Z] Warning: A secret was passed to "sh" using Groovy String interpolation, which is insecure.<br/>[2023-02-09T14:36:54.928Z] 		 Affected argument(s) used the following variable(s): [GITHUB_TOKEN]<br/>[2023-02-09T14:36:54.928Z] 		 See https://jenkins.io/redirect/groovy-string-interpolation for details.<br/>[2023-02-09T14:36:55.196Z] + build-chain build cross_pr --token **** -f https://raw.githubusercontent.com/kiegroup/kogito-pipelines/main/.ci/pull-request-config.yaml -o bc -p kiegroup/kogito-examples -u https://github.com/kiegroup/kogito-examples/pull/1561 --skipParallelCheckout<br/>[2023-02-09T14:36:55.196Z] [ERROR] Invalid event url https://github.com/kiegroup/kogito-examples/pull/1561<br/>[Pipeline] }<br/>[Pipeline] // withCredentials<br/>[Pipeline] }<br/>[2023-02-09T14:36:55.271Z] Deleting 1 temporary files<br/>[Pipeline] // configFileProvider<br/>[Pipeline] }<br/>[Pipeline] // script<br/>Post stage<br/>[Pipeline] junit<br/>[2023-02-09T14:36:55.372Z] Recording test results<br/>[2023-02-09T14:36:55.390Z] [Checks API] No suitable checks publisher found.<br/>[Pipeline] archiveArtifacts<br/>[2023-02-09T14:36:55.406Z] Archiving artifacts<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>[2023-02-09T14:36:55.801Z] + rm -rf console.log<br/>[Pipeline] sh<br/>
</details>

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Pull request merged**